File: C++/01-Arrays-Basics/1 Find Maximum Element.cpp
// Find Maximum Element in an Array
//example: arr[] = {10, 5, 8, 20, 3} => output: 20


#include <iostream>
using namespace std;

int findMax(int arr[], int n) {

    int max = arr[0];

    for (int i = 1; i < n; i++) {
        if (arr[i] > max) {
            max = arr[i];
        }
    }
    return max;
}

int main() {

    int arr[] = {10, 5, 8, 20, 3};

    int n = sizeof(arr) / sizeof(arr[0]);

    cout << "Maximum element is: " << findMax(arr, n) << endl;
    
    return 0;
}

File: Python/1.Arrays-Basics/1.Find Maximum Element.py
# Find the maximum element in an array

def find_max(arr):
    if not arr:
        return None
    max_element = arr[0]
    for num in arr:
        if num > max_element:
            max_element = num
    return max_element


array = [3, 5, 7, 2, 8]
print("Maximum element:", find_max(array))